controllers/mattermost/mattermost/mattermost.go

```diff
@@ -209,13 +219,38 @@ func (r *MattermostReconciler) checkMattermostIngress(mattermost *mmv1beta.Matte
 	return r.Resources.Update(current, desired, reqLogger)
 }
 
+func (r *MattermostReconciler) checkMattermostIngressClass(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) error {
+	var desired *networkingv1.IngressClass
+	desired = mattermostApp.GenerateALBIngressClassV1Beta(mattermost)
+
+	if mattermost.Spec.AWSLoadBalancerController == nil || mattermost.Spec.AWSLoadBalancerController.Enabled == false || mattermost.Spec.AWSLoadBalancerController.IngressClassName != "" {
```

Comment:
I think if `.IngressClassName == ""`, then you want to delete.

Reply:
If `.IngressClassName != ""` I need to delete the ingress class created by the operator. Since the Ingress will use the IngressClassName provided by the spec.